File: tests/testdata/codegen/kubernetes-pod-pp/python/kubernetes-pod.py
import pulumi
import pulumi_kubernetes as kubernetes

bar = kubernetes.core.v1.Pod("bar",
    api_version="v1",
    metadata={
        "namespace": "foo",
        "name": "bar",
        "labels": {
            "app.kubernetes.io/name": "cilium-agent",
            "app.kubernetes.io/part-of": "cilium",
            "k8s-app": "cilium",
        },
    },
    spec={
        "containers": [
            {
                "name": "nginx",
                "image": "nginx:1.14-alpine",
                "ports": [{
                    "container_port": 80,
                }],
                "resources": {
                    "limits": {
                        "memory": "20Mi",
                        "cpu": "0.2",
                    },
                },
            },
            {
                "name": "nginx2",
                "image": "nginx:1.14-alpine",
                "ports": [{
                    "container_port": 80,
                }],
                "resources": {
                    "limits": {
                        "memory": "20Mi",
                        "cpu": "0.2",
                    },
                },
            },
        ],
    })
# Test that we can assign from a constant without type errors
kind = bar.kind

File: tests/testdata/codegen/kubernetes-pod-pp/go/kubernetes-pod.go
package main

import (
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		bar, err := corev1.NewPod(ctx, "bar", &corev1.PodArgs{
			ApiVersion: "v1",
			Metadata: &*metav1.ObjectMetaArgs{
				Namespace: "foo",
				Name:      "bar",
				Labels: map[string]pulumi.String{
					"app.kubernetes.io/name":    pulumi.String("cilium-agent"),
					"app.kubernetes.io/part-of": pulumi.String("cilium"),
					"k8s-app":                   pulumi.String("cilium"),
				},
			},
			Spec: &*corev1.PodSpecArgs{
				Containers: corev1.ContainerArray{
					&corev1.ContainerArgs{
						Name:  pulumi.String("nginx"),
						Image: "nginx:1.14-alpine",
						Ports: []corev1.ContainerPortArgs{
							{
								ContainerPort: pulumi.Int(80),
							},
						},
						Resources: &*corev1.ResourceRequirementsArgs{
							Limits: map[string]pulumi.String{
								"memory": pulumi.String("20Mi"),
								"cpu":    pulumi.String("0.2"),
							},
						},
					},
					&corev1.ContainerArgs{
						Name:  pulumi.String("nginx2"),
						Image: "nginx:1.14-alpine",
						Ports: []corev1.ContainerPortArgs{
							{
								ContainerPort: pulumi.Int(80),
							},
						},
						Resources: &*corev1.ResourceRequirementsArgs{
							Limits: map[string]pulumi.String{
								"memory": pulumi.String("20Mi"),
								"cpu":    pulumi.String("0.2"),
							},
						},
					},
				},
			},
		})
		if err != nil {
			return err
		}
		// Test that we can assign from a constant without type errors
		_ := bar.Kind
		return nil
	})
}
